src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  padding: $calendar-padding;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: calendar-width($cell-size, $gap, $calendar-padding);
+
+  @each $name, $day in $startDay {
+    &--#{$name} &__day:first-child {
+      margin-left: $cell-size * $day + $gap * $day;
+    }
+  }
+
+  @for $days from 28 through 31 {
+    &--month-lendth--#{$days} :nth-child(n + #{$days + 1}) {
```

Comment:
There should not be two hyphens between `length`and the day count: `&--month-length-#{$day}`
And there is a typo in length